### PR TITLE
fix: Modal async

### LINKED ISSUE #23623: RangePicker受控模式行为异常
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-o4mui)

### Steps to reproduce
修改RangePicker时间范围，在Modal弹出时选择cancel
GIF 2020-4-26 22-00-17

### What is expected?
RangePicker控件显示的时间不变

### What is actually happening?
RangePicker控件显示的时间变了

| Environment | Info |
|---|---|
| antd | 4.1.5 |
| React | react  |
| System | windows |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #23623

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Adjust Modal.xxx function async to avoid block React events.       |
| 🇨🇳 Chinese |     调整 Modal.xxx 方法为异步以防止其影响 React 事件响应。    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/modal/confirm.tsx [hunk 1]
```diff
@@ -50,15 +50,21 @@ export default function confirm(config: ModalFuncProps) {
...
-        okText={okText || (props.okCancel ? runtimeLocale.okText : runtimeLocale.justOkText)}
-        cancelText={cancelText || runtimeLocale.cancelText}
-      />,
-      div,
-    );
+    /**
+     * https://github.com/ant-design/ant-design/issues/23623
+     * Sync render blocks React event. Let's make this async.
+     */
+    setTimeout(() => {
```
Comment (afc163) [inline components/modal/confirm.tsx:57]: 需要 clear 么？

Comment (zombieJ) [reply components/modal/confirm.tsx:57]: 不用，Modal.xxx 没有生命周期。clear 用的是 `Modal.destroyAll`

Comment (afc163) [reply components/modal/confirm.tsx:57]: 这么说 destroyAll 和 destroy 的时候是需要 clear 的，万一两个方法连续调用。

```
Modal.confirm();
Modal.destroyAll();
```

**Comment by zombieJ** [comment]
> 这个组件不用portal重写吗

静态方法是在 React 生命周期外的，没有调用 portal 的时机。

**Pull request merged**